plugins/recorded_future/komand_recorded_future/util/api.py

```diff
@@ -27,8 +28,30 @@ def setup_custom_header(self):
 
     def _call_api(self, method: str, endpoint: str, params: dict = None, data: dict = None, json: dict = None):
         _url = self.base_url + endpoint
+        try:
+            response = requests.request(
+                url=_url, method=method, params=params, data=data, json=json, headers=self.headers
+            )
+        except MemoryError:
+            raise PluginException(
+                cause="Memory Error: Returned dataset too large",
+                assistance="Please allow a larger amount of memory to parse this dataset, or try another list",
+                data=f"Memory Error using endpoint: {endpoint}",
+            )
+        self.response_handled(response, _url)
+        if endpoint.endswith("risklist"):
+            try:
+                decompressed_data = gzip.decompress(response.content)
+                return response.content, dict(xmltodict.parse(decompressed_data))
```

Comment:
would it better in this util function to do the `base64.b64encode(response_content).decode("utf-8")` and have an exception to catch any issue? and it'll save the duplicate code in each of the actions.

Reply:
Have moved this to the one location now.